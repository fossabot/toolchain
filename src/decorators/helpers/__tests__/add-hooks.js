import addHooks from '../add-hooks';

const callOrder = jest.fn();

describe('addHooks', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it('decorate input function with hooks and fire them in order', async () => {
    const callOrderWithName = hookName => () => {
      callOrder(hookName);
    };
    const beforeHook = callOrderWithName('beforeHook');
    const afterHook = callOrderWithName('afterHook');
    const original = callOrderWithName('original');

    const decorated = addHooks({ beforeHook, afterHook })(original);

    await decorated();

    expect(callOrder.mock.calls).toMatchSnapshot();
  });

  it('return the result of original function', async () => {
    const original = () => {
      return 'result';
    };

    const decorated = addHooks({})(original);

    const result = await decorated();

    expect(result).toBe('result');
  });
});
