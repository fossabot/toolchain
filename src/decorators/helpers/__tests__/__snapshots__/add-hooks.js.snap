// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addHooks decorate input function with hooks and fire them in order 1`] = `
Array [
  Array [
    "beforeHook",
  ],
  Array [
    "original",
  ],
  Array [
    "afterHook",
  ],
]
`;
